### PR TITLE
upgraded to webpack 2

### PR DESCRIPTION
**Pull request by rfreitas**
### Scope

Upgraded to webpack 2, since there is already a webpack 3 out

### Other Fixes

1. Refactoring of the config file
2. Turned off default minification in favour of one controlled by the config file
3. Turned off minification for the npm output bundle

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://171-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://171-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://171-pr-onfido-sdk-ui-onfido.surge.sh

File: webpack.config.babel.js [hunk 1]
```diff
@@ -7,170 +7,191 @@ import url from 'postcss-url';
...
+const baseRules = [{
   test: /\.jsx?$/,
-  include: [`${__dirname}/src`, `${__dirname}/test`],
-  loader: 'babel-loader'
+  include: [
+    `${__dirname}/src`,
+    /*
+    *  Necessary because preact-compat": "3.4.2" has babel in it,
+    *  so webpack2 crashes on UglifyJsPlugin step
+    *  see: https://github.com/developit/preact-compat/issues/155
```
Comment (stephencookdev) [inline webpack.config.babel.js]: That's annoying :/

Comment (rfreitas) [inline webpack.config.babel.js]: yup :/ but overraly the webpack 2 schema is much less hacky. I welcome the strictness of it as well, build script tend to get redundant without it.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://171-pr-onfido-sdk-ui-onfido.surge.sh

File: webpack.config.babel.js (hunk 1 above)
Comment (TheComfyChair) [inline webpack.config.babel.js]: Also, the merge functionality is great. Makes creating a bunch of environment dependant config a breeze.

File: webpack.config.babel.js [hunk 2]
```diff
@@ -7,170 +7,191 @@ import url from 'postcss-url';
...
-    process: false,
-    Buffer: false,
-    __filename: false,
-    __dirname: false,
-    setImmediate: false
-  },
-
-  devtool: ENV==='production' ? 'source-map' : 'eval-source-map',
+    }),
+    ... ENV === 'production' ?
```
Comment (TheComfyChair) [inline webpack.config.babel.js]: The webpack merge utility introduced in v2 is great for handling differences between different environments, so I'd recommend looking into adding it here :)

**Comment by dawiddylowicz** [comment]
build passes, tests passing -> merging

File: webpack.config.babel.js (hunk 1 above)
Comment (rfreitas) [inline webpack.config.babel.js]: this: https://github.com/survivejs/webpack-merge ?
It seems it does a deep merge, looks cool!

**Pull request merged**